### PR TITLE
Live Location Sharing - Update beacon info state event when sharing is ended

### PR DESCRIPTION
**Pull request by onurays**

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt [hunk 1]
```diff
@@ -140,6 +145,25 @@ class LocationSharingService : VectorService(), LocationTracker.Callback {
         }
     }
 
+    private fun updateStoppedBeaconInfo(roomId: String) {
```
Comment (mnaturel) [inline vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt:148]: To clarify namings what do you think about renaming this method to `sendStoppedBeaconInfo()` and the method `sendBeaconInfo()` to `sendLivedBeaconInfo()` ?

File: vector/src/main/java/im/vector/app/features/location/LocationSharingServiceConnection.kt [hunk 2]
```diff
@@ -51,7 +52,12 @@ class LocationSharingServiceConnection @Inject constructor(
         callback = null
     }
 
+    fun stopLiveLocationSharing(roomId: String) {
+        locationSharingService?.stopSharingLocation(roomId)
+    }
+
     override fun onServiceConnected(className: ComponentName, binder: IBinder) {
+        locationSharingService = (binder as LocationSharingService.LocalBinder).getService()
```
Comment (mnaturel) [inline vector/src/main/java/im/vector/app/features/location/LocationSharingServiceConnection.kt:60]: Should we reset `locationSharingService` to `null` in `onServiceDisconnected` for safety reason?

**Review by mnaturel — COMMENTED** [review]
I have just added small comments to improve code.

File: vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt [hunk 3]
```diff
@@ -140,6 +145,25 @@ class LocationSharingService : VectorService(), LocationTracker.Callback {
...
+        activeSessionHolder
+                .getSafeActiveSession()
+                ?.let { session ->
+                    session.coroutineScope.launch(session.coroutineDispatchers.io) {
+                        val room = session.getRoom(roomId)
+                        EventType
```
Comment (mnaturel) [inline vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt:154]: I think we also need to add a filter to get all state events with `lived=true`. Normally there should be only a single live beacon per user per room. Maybe we need to extract it into a dedicated method to use it also in `sendLiveLocation`. I see in this last method 2 missing filters: the state_key=userId and lived=true which will be solved by this new method.

File: vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt (hunk 1 above)
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt]: Done.

File: vector/src/main/java/im/vector/app/features/location/LocationSharingServiceConnection.kt (hunk 2 above)
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/location/LocationSharingServiceConnection.kt:60]: Good catch, done.

File: vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt (hunk 3 above)
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt]: Moved to an helper function.

**Pull request merged**